FE Board UI

ahmed-belhadj opened the pull request:
### What this PR does (required):
- [x] Use mock data, no integration is needed for now
- [x] No top navbar
- [x] Cards should be draggable within a column and between them
- [x] Column should be draggable

- [x] Uses functional components
- [x] Uses MUI's styling solution

### Screenshots / Videos (required):
- https://www.loom.com/share/93fb873ccb384b458f21f428bc9e317f

### Any information needed to test this feature (required):
- http://localhost:3000/board-ui

### Any issues with the current functionality (optional):
- ~Use functional components.~
- ~Use of styled-components here does not use MUI's styling solution but it is still compatible.~
- Refactor 4 remaining files into Typescript.

felix-quinn reviewed (approved):
This works as expected on my computer. No notable errors detected.

samimsu reviewed (approved):
Works amazingly!

bonnieli commented on client/src/App.tsx:
```diff
@@ -4,7 +4,8 @@ import { BrowserRouter, Route, Redirect, Switch } from 'react-router-dom';
 import Dashboard from './pages/Dashboard/Dashboard';
-import { AuthProvider } from './context/useAuthContext';
+import BoardUI from './components/BoardUI/BoardUI';
+// import { AuthProvider } from './context/useAuthContext';
```
commented out code here?

ahmed-belhadj replied:
Commented out AuthProvider to prevent re-directs to Login page so that team can view Board component.

Should I add it back?

sina-jamshidi replied:
When pushing code to main we want to avoid commented out code. If it needs to be commented it can be removed :) We can always add it back in later if we need to.
In this case I would say you should uncomment useAuth. Remember that main is where all your teammates will also be pulling from, so we want to make sure it is the most up to date running version of our app!

bonnieli commented on client/src/components/BoardUI/BoardUI.js:
```diff
@@ -0,0 +1,137 @@
+  }
+}
+
+export default class BoardUI extends React.Component {
```
Let's convert this into a functional component

bonnieli commented on client/src/components/BoardUI/BoardUI.js:
```diff
@@ -0,0 +1,137 @@
+  display: flex;
+`;
+
+class InnerList extends React.PureComponent {
```
Let's convert this into a functional component

bonnieli commented on client/src/components/BoardUI/Card.js:
```diff
@@ -0,0 +1,32 @@
+import styled from 'styled-components';
+import { Draggable } from 'react-beautiful-dnd';
+
+const Container = styled.div`
```
we're using styled-components here? This is okay - but we'll have to stay consistent across the entire app. may want to check in with the team this is the direction we'd like to go

ahmed-belhadj replied:
Agreed, checked in with team and working on implementing MUI's styling solution.

bonnieli reviewed (commented):
Great start, nice clean code. 

I would convert all the components into functional components, and also sync with the team about using styled-components

ahmed-belhadj committed:
Update package.json

Upgrade react-beautiful-dnd
Install natural-drag-animation-rbdnd for smooth card animations
Uninstall styled-components

ahmed-belhadj committed:
Update package-lock.json

ahmed-belhadj committed:
Delete mock-data.js

ahmed-belhadj committed:
Create mock-data.ts

ahmed-belhadj committed:
Create withDroppable.tsx

wrapper for droppable components

ahmed-belhadj committed:
Create useStyles.ts

board styles

ahmed-belhadj committed:
Delete Card.js

ahmed-belhadj committed:
Create Card.tsx

ahmed-belhadj committed:
Create DefaultCard.tsx

ahmed-belhadj committed:
board helper functions

ahmed-belhadj committed:
Delete Column.js

ahmed-belhadj committed:
Create Column.tsx

ahmed-belhadj committed:
Create ColumnHeader.tsx

ahmed-belhadj committed:
Create Board.tsx

ahmed-belhadj committed:
Update BoardUI.js

ahmed-belhadj committed:
Update App.tsx

ahmed-belhadj committed:
Update react-app-env.d.ts

ahmed-belhadj committed:
Merge branch 'main' into fe-board-ui

ahmed-belhadj committed:
Revert "Merge branch 'main' into fe-board-ui"

This reverts commit c6656f79e89840daddae5aae40b00c5edc9749d2, reversing
changes made to f95bb3919c2a86c0246a059604ea77c153b37eaa.

ahmed-belhadj committed:
Revert "Revert "Merge branch 'main' into fe-board-ui""

This reverts commit daffb806551306b1ae98d3d73de8a8f0cfdfa2ee.
